homeassistant/components/incomfort/water_heater.py

```diff
@@ -0,0 +1,96 @@
+"""Support for an Intergas boiler via an InComfort/Intouch Lan2RF gateway."""
+import asyncio
+import logging
+
+from homeassistant.components.water_heater import WaterHeaterDevice
+from homeassistant.const import TEMP_CELSIUS
+
+from . import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+INCOMFORT_SUPPORT_FLAGS = 0
+
+INCOMFORT_MAX_TEMP = 80.0
+INCOMFORT_MIN_TEMP = 30.0
+
+
+async def async_setup_platform(hass, hass_config, async_add_entities,
+                               discovery_info=None):
+    """Set up an InComfort/Intouch water_heater device."""
+    client = hass.data[DOMAIN]['client']
+    heater = hass.data[DOMAIN]['heater']
+
+    async_add_entities([
+        IncomfortWaterHeater(client, heater)], update_before_add=True)
+
+
+class IncomfortWaterHeater(WaterHeaterDevice):
+    """Representation of an InComfort/Intouch water_heater device."""
+
+    def __init__(self, client, boiler):
+        """Initialize the water_heater device."""
+        self._client = client
+        self._boiler = boiler
+        self._name = 'Boiler'
+
+    @property
+    def name(self):
+        """Return the name of the water_heater device."""
+        return self._name
+
+    @property
+    def device_state_attributes(self):
+        """Return the device state attributes."""
+        keys = ['nodenr', 'rf_message_rssi', 'rfstatus_cntr', 'room_1',
+                'room_2']
+        state = {k: self._boiler.status[k]
+                 for k in self._boiler.status if k not in keys}
```

Comment:
What items will be included?

Reply:
Here's an example:
```python
{'display_code': 126, 'display_text': 'standby', 'fault_code': 0, 'is_burning': False, 'is_failed': False, 'is_pumping': False, 'is_tapping': False, 'heater_temp': 57.4, 'tap_temp': 50.61, 'pressure': 1.29, 'serial_no': '179t23082'}
```

If used `not in` because it was shorter.  Can change to `in` if you require.

Reply:
It's good to be able to see what attributes we report as state I would recommend that.

Serial no is not allowed in state attributes. We only want dynamic info that is related to the state of the device. Serial no can go into device info when there is a config entry for the integration (not at this point).

Reply:
changed to `in HEATER_ATTRS`, after removing serial number.